class.jetpack.php

```diff
@@ -6027,4 +6052,53 @@ public function dashboard_widget_connect_to_wpcom() {
 		<?php
 	}
 
+	/*
+	 * A graceful transition to using Core's site icon.
+	 *
+	 * All of the hard work has already been done with the image
+	 * in all_done_page(). All that needs to be done now is update
+	 * the option and display proper messaging.
+	 *
+	 * @todo remove when WP 4.3 is minimum
+	 *
+	 * @since 3.6.1
+	 *
+	 * @return bool false = Core's icon not available || true = Core's icon is available
+	 */
+	public static function jetpack_site_icon_available_in_core() {
+		global $wp_version;
+		$core_icon_available = function_exists( 'has_site_icon' ) && version_compare( $wp_version, '4.3-beta' ) >= 0;
+
+		if ( ! $core_icon_available ) {
+			return false;
+		}
+
+		// No need for Jetpack's site icon anymore if core's is already set
```

Comment:
I think this part doesn't make so much sense to me. 
Since we should just not load the module anymore if the user is running 3.6 not just if the user has an site icon or not. 

And this part needs to be tested from both ways. If the user is running jetpack WP 4.2 and then upgrade to WP 4.3 and the latest jetpack or if the user is on jetpack 3.5 and upgrades to jetpack 3.6 and is running WP 4.3.